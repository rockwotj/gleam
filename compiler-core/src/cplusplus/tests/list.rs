use crate::assert_cpp;

#[test]
fn make_list() {
    assert_cpp!(
        r#"
pub fn palindrome(x: String, y: String, z: String) -> List(String) {
    [x, y, z, y, x]
}"#
    );
}
