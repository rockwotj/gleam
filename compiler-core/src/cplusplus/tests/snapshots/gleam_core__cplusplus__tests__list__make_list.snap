---
source: compiler-core/src/cplusplus/tests/list.rs
expression: "\npub fn palindrome(x: String, y: String, z: String) -> List(String) {\n    [x, y, z, y, x]\n}"
---
#ifndef MY_PACKAGE_MY_MODULE_H_
#define MY_PACKAGE_MY_MODULE_H_

#include <gleam.h>

namespace my {
namespace module {

gleam::Ref<gleam::List<gleam::String>> palindrome(gleam::String x, gleam::String y, gleam::String z);

} // namespace my
} // namespace module

#endif // MY_PACKAGE_MY_MODULE_H_

---
#include "module.h"

namespace my {
namespace module {

gleam::Ref<gleam::List<gleam::String>> palindrome(gleam::String x, gleam::String y, gleam::String z) {
  return gleam::MakeList<gleam::String>({x, y, z, y, x});
};

} // namespace my
} // namespace module

